source/gameengine/GameLogic/Joystick/SCA_Joystick.cpp

```diff
@@ -200,119 +185,151 @@ bool SCA_Joystick::aAxisIsPositive(int axis_single)
 bool SCA_Joystick::aAnyButtonPressIsPositive(void)
 {
 #ifdef WITH_SDL
-	if (!SDL_CHECK(SDL_JoystickGetButton)) {
+	if (!(SDL_CHECK(SDL_GameControllerGetButton))) {
 		return false;
 	}
+
 	/* this is needed for the "all events" option
 	 * so we know if there are no buttons pressed */
-	for (int i=0; i<m_buttonmax; i++)
-		if (SDL_JoystickGetButton(m_private->m_joystick, i))
+	for (int i = 0; i < m_buttonmax; i++) {
+		if (SDL_GameControllerGetButton(m_private->m_gamecontroller, (SDL_GameControllerButton)i)) {
 			return true;
+		}
+	}
 #endif
 	return false;
 }
 
 bool SCA_Joystick::aButtonPressIsPositive(int button)
 {
-#ifndef WITH_SDL
-	return false;
-#else
-	bool result;
-	result = SDL_CHECK(SDL_JoystickGetButton) && SDL_JoystickGetButton(m_private->m_joystick, button);
-	return result;
+#ifdef WITH_SDL
+	if ((SDL_CHECK(SDL_GameControllerGetButton) &&
+	     SDL_GameControllerGetButton(m_private->m_gamecontroller, (SDL_GameControllerButton)button)))
+	{
+		return true;
+	}
 #endif
+	return false;
 }
 
 
 bool SCA_Joystick::aButtonReleaseIsPositive(int button)
 {
-#ifndef WITH_SDL
-	return false;
-#else
-	bool result;
-	result = !(SDL_CHECK(SDL_JoystickGetButton) && SDL_JoystickGetButton(m_private->m_joystick, button));
-	return result;
+#ifdef WITH_SDL
+	if (!(SDL_CHECK(SDL_GameControllerGetButton) &&
+       SDL_GameControllerGetButton(m_private->m_gamecontroller, (SDL_GameControllerButton)button)))
```

Comment:
For code style it would be nice to indent this line, or align it with the first "(".